x-pack/packages/ai-infra/inference-common/src/output/api.ts

```diff
@@ -6,39 +6,141 @@
  */
 
 import type { Observable } from 'rxjs';
-import type { Message, FunctionCallingMode, FromToolSchema, ToolSchema } from '../chat_complete';
-import type { OutputEvent } from './events';
+import { Message, FunctionCallingMode, FromToolSchema, ToolSchema } from '../chat_complete';
+import { Output, OutputEvent } from './events';
 
 /**
  * Generate a response with the LLM for a prompt, optionally based on a schema.
  *
- * @param {string} id The id of the operation
- * @param {string} options.connectorId The ID of the connector that is to be used.
- * @param {string} options.input The prompt for the LLM.
- * @param {string} options.messages Previous messages in a conversation.
- * @param {ToolSchema} [options.schema] The schema the response from the LLM should adhere to.
+ * @example
+ * ```ts
+ * // schema must be defined as full const or using the `satisfies ToolSchema` modifier for TS type inference to work
+ * const mySchema = {
+ *   type: 'object',
+ *   properties: {
+ *     animals: {
+ *       description: 'the list of animals that are mentioned in the provided article',
+ *       type: 'array',
+ *       items: {
+ *         type: 'string',
+ *       },
+ *     },
+ *   },
+ * } as const;
+ *
+ * const response = outputApi({
+ *   id: 'extract_from_article',
+ *   connectorId: 'my-connector connector',
+ *   schema: mySchema,
+ *   input: `
+ *     Please find all the animals that are mentioned in the following document:
+ *     ## Document¬
+ *     ${theDoc}
+ *   `,
+ * });
+ *
+ * // output is properly typed from the provided schema
+ * const { animals } = response.output;
+ * ```
  */
 export type OutputAPI = <
   TId extends string = string,
-  TOutputSchema extends ToolSchema | undefined = ToolSchema | undefined
+  TOutputSchema extends ToolSchema | undefined = ToolSchema | undefined,
+  TStream extends boolean = false
 >(
-  id: TId,
-  options: {
-    connectorId: string;
-    system?: string;
-    input: string;
-    schema?: TOutputSchema;
-    previousMessages?: Message[];
-    functionCalling?: FunctionCallingMode;
-  }
-) => OutputResponse<TId, TOutputSchema>;
+  options: OutputOptions<TId, TOutputSchema, TStream>
+) => OutputCompositeResponse<TId, TOutputSchema, TStream>;
+
+/**
+ * Options for the {@link OutputAPI}
+ */
+export interface OutputOptions<
+  TId extends string = string,
+  TOutputSchema extends ToolSchema | undefined = ToolSchema | undefined,
+  TStream extends boolean = false
+> {
+  /**
+   * The id of the operation.
+   */
+  id: TId;
+  /**
+   * The ID of the connector to use.
+   * Must be a genAI compatible connector, or an error will be thrown.
+   */
+  connectorId: string;
+  /**
+   * Optional system message for the LLM.
+   */
+  system?: string;
+  /**
+   * The prompt for the LLM.
+   */
+  input: string;
+  /**
+   * The {@link ToolSchema} the response from the LLM should adhere to.
```

Comment:
`TOutputSchema` appears to `extend ToolSchema` without augmenting it, so is it sufficient to rely on the typing for `TOutputSchema`, or is the `@link` reference in the doc providing another benefit?

```suggestion
   * The response format the LLM should adhere to.
```

Reply:
I usually link to the corresponding root generic type in generic/parametrized property's doc, just because neither vscode nor intellij does it (can't one-click access the generic type from "show doc/show def" on neither of the 2 mostly used IDE - afaik). But that's basically it, I don't mind removing it.